aea/protocols/dialogue/base.py

```diff
@@ -699,7 +705,7 @@ def reply(
 
         :return: the reply message if it was successfully added as a reply, None otherwise.
         """
-        last_message = self.last_message
+        last_message = self.last_incoming_message
```

Comment:
This should not be changed. The last message != last_incoming_message in general.